airflow/contrib/hooks/aws_athena_hook.py

```diff
@@ -148,3 +156,105 @@ def stop_query(self, query_execution_id):
         :return: dict
         """
         return self.conn.stop_query_execution(QueryExecutionId=query_execution_id)
+
+
+class AWSAthenaHelpers(AWSAthenaHook):
+    """
+    The Athena Helpers contains helper methods to execute queries against
+    Athena. The methods can be used directly by operators.
+    """
+
+    def __init__(self, aws_conn_id='aws_default', region_name=None, *args, **kwargs):
+        super(AWSAthenaHelpers, self).__init__(
+            aws_conn_id=aws_conn_id, region_name=region_name, **kwargs)
+        self.region_name = region_name
+        self.s3_hook = None
+        self.glue_hook = None
+
+    def get_s3_hook(self):
+        """
+        check if s3 hook exists already or create one and return it
+        :return: s3 hook
+        """
+        if not self.s3_hook:
+            self.s3_hook = S3Hook(
+                aws_conn_id=self.aws_conn_id, verify=self.verify)
+        return self.s3_hook
+
+    def get_glue_hook(self):
+        """
+        check if glue hook exists already or create one and return it
+        :return: glue hook
+        """
+        if not self.glue_hook:
+            self.glue_hook = AwsGlueCatalogHook(
+                aws_conn_id=self.aws_conn_id, region_name=self.region_name)
+        return self.glue_hook
+
+    def run_insert_into_table(self, src_db, src_table, dst_db, dst_table, mode='error'):
+        """
+        insert data in s3 from the source table to the destination table
```

Comment:
Why is this going via S3 rather than issuing an athena query?

Reply:
Because original Athena didn't support `INSERT INTO TABLE AS SELECT` clause, so actually we need to `create a temp table as select` and move data from the location of temp table to the s3 location of the target table.

Reply:
@bryanyang0528 
Can you please explain the use case for this? 

When I want to add data to Athena I just upload files to the same directory and it works like a charm as Athena scans the whole folder.
For cases where I upload files to another path I just need to update the table partition with:
`ALTER TABLE _____ADD IF NOT EXISTS PARTITION (dt=____) location 's3://____`
https://stackoverflow.com/questions/50164744/how-to-efficiently-append-new-data-to-table-in-aws-athena

Reply:
@rosner  Thank you for your suggestion. I knew `ATLER TABLE` can assign an s3 location for partitions. But I would like to move data to the same s3 location after creating a new partition.

For example, I have a table called `user_profile` in which I aggregated user behavior data on my website. I have to select data from the source table `daily` and insert the result to  `user_profile`. In this case, in Hive, I just use `insert into table user_profile partition (dt) as select count(*) from src_table`.   And data will be put into the same location as the target table.

Maybe I could combine `Alter table` and `s3 copy` to implement this idea. Does it make sense?

Reply:
@bryanyang0528 better to wait for @ashb respond.

Reply:
@RosterIn Thx.

Reply:
If (and it is still and if) we add this to Airflow this function belongs in an Operator, not a Hook.

Additionally you are putting S3 files in place but doing nothing to update the Glue catalog.

Overall I am skeptical that this approach is the right way of doing it. In the past I have done this sort of thing from an EMR cluster using the Glue-compatible hive metastore to insert into tables.

Reply:
Thank you for the suggestion. I will close this issue and create a new one. I know EMR could do that thing, but it takes a long time to create an EMR cluster and compute on the EMR.